api_tests/src/wallets/bitcoind_wallet.ts

```diff
@@ -0,0 +1,88 @@
+import { BitcoinWallet } from "comit-sdk";
+import { BitcoinNodeConfig } from "../ledgers";
+import BitcoinRpcClient from "bitcoin-core";
+
+export class BitcoindWallet implements BitcoinWallet {
+    constructor(private readonly client: BitcoinRpcClient) {}
+
+    public static async newInstance(
+        network: string,
+        hdKey: string,
+        config: BitcoinNodeConfig
+    ): Promise<BitcoindWallet> {
+        const bitcoinRpcClient = new BitcoinRpcClient({
+            network,
+            host: "localhost",
+            port: config.rpcPort,
+            username: config.username,
+            password: config.password,
+            wallet: "miner",
+        });
+
+        // everything before is the same for every xprv
+        const walletName = hdKey.substr(16, 20);
```

Comment:
Can you clarify why you are doing this?

Reply:
I would suggest to either:

- use a library for parsing this that makes it more readable
- include a code comment of the structure of an hdkey that describes what you are doing (maybe including an example what an hdkey looks like?)

Reply:
What do you want to use a library for? 

Maybe it's better to just use Math.random() for the name as proposed by @luckysori :D